### PR TITLE
[pkg/teststeps] Implement ForEachNonResumingTarget helper function and use it in cmd plugin

### PR DESCRIPTION
**Pull request by rihter007**
Introduce ForEachNonResumingTarget helper function that provides easier functionality for plugins that do not support pause/restore.

It is also useful if plugin internally uses a blocking operation that takes context.Context as its input
Rewrite cmd plugin using the new ForEachNonResumingTarget

### CODE REVIEW

File: plugins/teststeps/teststeps.go [hunk 1]
```diff
@@ -120,3 +121,38 @@ func ForEachTarget(pluginName string, cancel, pause <-chan struct{}, ch test.Tes
...
+	defer cancelFunc()
+
+	return ForEachTarget(pluginName, cancel, pause, ch, func(cancel, pause <-chan struct{}, target *target.Target) error {
+		return f(ctx, target)
+	})
+}
+
+func newTerminateContext(cancel, pause <-chan struct{}) (context.Context, context.CancelFunc) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	go func() {
```
Comment (rihter007) [inline plugins/teststeps/teststeps.go:148]: This is so bad to create an extra goroutine, I wish it could be possible to override <- in golang

**Comment by rihter007** [comment]
Of the concept is ok, I will make some unit tests.

File: plugins/teststeps/teststeps.go (hunk 1 above)
Comment (xaionaro) [reply plugins/teststeps/teststeps.go:148]: > so bad

Out of curiosity: why?

**Review by xaionaro — APPROVED** [review]
LGTM. And I like the approach of converting signal channels to a context. I was hoping ConTest will start moving to this direction :)

File: plugins/teststeps/teststeps.go (hunk 1 above)
Comment (rihter007) [reply plugins/teststeps/teststeps.go:148]: Some extra overhead instead of <- overriding

**Comment by rihter007** [comment]
Is obsolete, we agreed to use statectx.Context

**Pull request closed without merging**